galsim/image.py

```diff
@@ -1171,30 +1214,58 @@ def addValue(self, *args, **kwargs):
         """
         if self.isconst:
             raise ValueError("Cannot modify the values of an immutable Image")
+        if not self.bounds.isDefined():
+            raise RuntimeError("Attempt to set value of an undefined image")
         pos, value = galsim.utilities.parse_pos_args(args, kwargs, 'x', 'y', integer=True,
                                                      others=['value'])
-        self.image.setValue(pos.x, pos.y, value + self.image(pos.x,pos.y))
+        if not self.bounds.includes(pos):
+            raise RuntimeError("Attempt to set position %s, not in bounds %s"%(pos,self.bounds))
+        self._addValue(pos.x,pos.y,value)
+
+    def _addValue(self, x, y, value):
+        """Equivalent to self.addValue(x,y,value) except that there are no checks that the values
+        fall within the bounds of the image.
+        """
+        self._array[y-self.ymin, x-self.xmin] += value
 
     def fill(self, value):
         """Set all pixel values to the given `value`
         """
         if self.isconst:
             raise ValueError("Cannot modify the values of an immutable Image")
-        self.image.fill(value)
+        if not self.bounds.isDefined(): return
```

Comment:
Shouldn't there be a warning or something?  (Similar to what happens in `invertSelf` when bounds are not defined)